### PR TITLE
[PREGEL] s/std::(unordered_map|unordered_set|map|set)/containers::FlatHash(Map…

### PR DESCRIPTION
**Pull request by markuspf**
Use abseil containers (almost) throughout.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Pregel/IncomingCache.h [hunk 1]
```diff
@@ -47,7 +47,9 @@ processing */
 template<typename M>
 class InCache {
  protected:
-  mutable std::map<PregelShard, std::mutex> _bucketLocker;
+  mutable std::unordered_map<PregelShard, std::mutex> _bucketLocker;
+  // mutable std::unordered_mapcontainers::FlatHashMap<PregelShard, std::mutex>
+  // _bucketLocker;
```
Comment (MBkkt) [inline arangod/Pregel/IncomingCache.h:52]:
```suggestion
  mutable containers::NodeHashMap<PregelShard, std::mutex> _bucketLocker;
```

File: arangod/Pregel/Algos/DMID/VertexSumAggregator.h [hunk 2]
```diff
@@ -38,9 +38,9 @@ namespace arangodb {
 namespace pregel {
 
 struct VertexSumAggregator : public IAggregator {
-  typedef std::map<PregelShard, std::unordered_map<std::string, double>>
+  typedef std::map<PregelShard, containers::FlatHashMap<std::string, double>>
       VertexMap;
```
Comment (MBkkt) [inline arangod/Pregel/Algos/DMID/VertexSumAggregator.h:42]: Why VertexMap should be ordered?

File: arangod/Pregel/IncomingCache.h (hunk 1 above)
Comment (markuspf) [reply arangod/Pregel/IncomingCache.h:52]: Why?

File: arangod/Pregel/Algos/DMID/VertexSumAggregator.h (hunk 2 above)
Comment (markuspf) [reply arangod/Pregel/Algos/DMID/VertexSumAggregator.h:42]: Why not?

**Pull request closed without merging**